controller/state.go

```diff
@@ -162,22 +166,23 @@ func (m *appStateManager) getRepoObjs(app *v1alpha1.Application, source v1alpha1
 	}
 	ts.AddCheckpoint("version_ms")
 	manifestInfo, err := repoClient.GenerateManifest(context.Background(), &apiclient.ManifestRequest{
-		Repo:              repo,
-		Repos:             permittedHelmRepos,
-		Revision:          revision,
-		NoCache:           noCache,
-		NoRevisionCache:   noRevisionCache,
-		AppLabelKey:       appLabelKey,
-		AppName:           app.Name,
-		Namespace:         app.Spec.Destination.Namespace,
-		ApplicationSource: &source,
-		Plugins:           tools,
-		KustomizeOptions:  kustomizeOptions,
-		KubeVersion:       serverVersion,
-		ApiVersions:       argo.APIResourcesToStrings(apiResources, true),
-		VerifySignature:   verifySignature,
-		HelmRepoCreds:     permittedHelmCredentials,
-		TrackingMethod:    string(argo.GetTrackingMethod(m.settingsMgr)),
+		Repo:                                  repo,
+		Repos:                                 permittedHelmRepos,
+		Revision:                              revision,
+		NoCache:                               noCache,
+		NoRevisionCache:                       noRevisionCache,
+		AppLabelKey:                           appLabelKey,
+		AppName:                               app.Name,
+		Namespace:                             app.Spec.Destination.Namespace,
+		ApplicationSource:                     &source,
+		Plugins:                               tools,
+		KustomizeOptions:                      kustomizeOptions,
+		KubeVersion:                           serverVersion,
+		ApiVersions:                           argo.APIResourcesToStrings(apiResources, true),
+		VerifySignature:                       verifySignature,
+		HelmRepoCreds:                         permittedHelmCredentials,
+		TrackingMethod:                        string(argo.GetTrackingMethod(m.settingsMgr)),
+		EnableManifestGenerationForSourceType: enableGeneration,
```

Comment:
nit: EnableManifestGenerationForSourceType. Can we shorten the field name? I suggest:

`EnabledSourceTypes`

Reply:
I like the suggestion. Thank you! Applied.